Add Vendor Detail v2 Api

poudelprakash opened the pull request:
- Add V2 Vendor Detail Api
- Add relevant tests

damianlegawiec commented on app/serializers/spree/v2/storefront/vendor_serializer.rb:
```diff
@@ -4,7 +4,10 @@ module Storefront
         set_type :vendor
 
-        attributes :name, :about_us
+        attributes :name, :about_us, :slug, :state, :contact_us
```
@poudelprakash let's not include state here

poudelprakash replied:
@damianlegawiec Done, State has been removed

damianlegawiec commented on app/serializers/spree/v2/storefront/vendor_serializer.rb:
```diff
@@ -4,7 +4,10 @@ module Storefront
-        attributes :name, :about_us
+        attributes :name, :about_us, :slug, :state, :contact_us
+
+        has_one :image
```
@poudelprakash we need to declare a serializer for this

poudelprakash replied:
@damianlegawiec it takes https://github.com/spree/spree/blob/master/api/app/serializers/spree/v2/storefront/image_serializer.rb
from the base,
any reason for this suggestion?

damianlegawiec replied:
@poudelprakash it should work, but we use a different class here: https://github.com/spree-contrib/spree_multi_vendor/blob/master/app/models/spree/vendor_image.rb

poudelprakash replied:
I have now added serializer and spec for vendor_image, please check :)

damianlegawiec commented on app/serializers/spree_multi_vendor/v2/storefront/product_serializer_decorator.rb:
```diff
@@ -0,0 +1,13 @@
+module SpreeMultiVendor
```
@poudelprakash we don't need this decorator anymore, already done

poudelprakash replied:
I don't understand where it is already done..
Is it already merged?

damianlegawiec replied:
> I don't understand where it is already done..
> Is it already merged?

https://github.com/spree-contrib/spree_multi_vendor/blob/9619f3a68cb253c53d2ba7b697267e2d93c0ca47/app/serializers/spree_multi_vendor/vendor_models_decorator.rb#L1

poudelprakash replied:
Awesome!

I have removed the decorator :)

damianlegawiec commented on app/controllers/spree/api/v2/storefront/vendors_controller.rb:
```diff
@@ -0,0 +1,27 @@
+          private
+
+          def scope
+            Spree::Vendor
```
```suggestion
            ::Spree::Vendor.active
```

poudelprakash replied:
Done

poudelprakash committed:
Vendor Detail Api
- Add Routes, Controller
- Add Test cases
- Update serializer with these attributes
  *slug
  *contact_us

Remove v2/storefront/product_serializer_decorator.rb

Add vendor image serializer and rspec

damianlegawiec commented on spec/requests/spree/api/v2/storefront/vendors_spec.rb:
```diff
@@ -0,0 +1,44 @@
+
+      it_behaves_like 'returns 200 HTTP status'
+
+      it 'returns product information' do
```
@poudelprakash let's also add a check for image attributes, thanks!

poudelprakash replied:
I have now added a check for image attributes as well, [Please check here](https://github.com/spree-contrib/spree_multi_vendor/pull/156/files#diff-7bce2b451c4453d90192df7319a3e5f45ab01b9788f8ab578e24e9dac2847504R44) :)

poudelprakash commented on spec/factories/spree/vendor_image_factory.rb:
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :vendor_image, class: Spree::VendorImage do
+    attachment { Rack::Test::UploadedFile.new("#{Rails.root}/../fixtures/thinking-cat.jpg", 'image/jpeg') }
```
@damianlegawiec I am new to image testing for engine,
Is there any better way to specify this path?
I had to use `Rails.root` to get dummy project path, and go one step outside with: `../` to reach fixtures folder.

Else the file would not be found.

Thank you :)

damianlegawiec replied:
https://github.com/spree/spree/blob/fd0131aea17c3a25972cf5079288bec7bd6870bc/backend/spec/controllers/spree/admin/stores_controller_spec.rb#L7 probably sth like this :)

damianlegawiec commented on app/serializers/spree/v2/storefront/vendor_image_serializer.rb:
```diff
@@ -0,0 +1,11 @@
+      class VendorImageSerializer < BaseSerializer
+        set_type :vendor_image
+
+        attributes :viewable_type, :viewable_id, :styles
```
@poudelprakash we can skip type/id

```suggestion
        attributes :styles
```

damianlegawiec reviewed (approved):
Thank you @poudelprakash , awesome work!
